app/code/Magento/Sales/Model/Order/Shipment.php

```diff
@@ -123,6 +143,8 @@ public function __construct(
         \Magento\Sales\Model\Order\Shipment\CommentFactory $commentFactory,
         \Magento\Sales\Model\ResourceModel\Order\Shipment\Comment\CollectionFactory $commentCollectionFactory,
         \Magento\Sales\Api\OrderRepositoryInterface $orderRepository,
+        SearchCriteriaBuilder $searchBuilder,
```

Comment:
New constructor parameters of existing classes shoud be backward compatible.

Example:
```
__constructor(
...
 SearchCriteriaBuilder $searchBuilder = null
...
) {
...
$this->searchBuilder = $searchBuilder ?: ObjectManager::getInstance()->get(SearchCriteriaBuilder::class);
...
```